src/oc/web/actions/activity.cljs

```diff
@@ -338,10 +338,12 @@
   (dis/dispatch! [:entry-modal-save]))
 
 (defn add-attachment [dispatch-input-key attachment-data]
-  (dis/dispatch! [:activity-add-attachment dispatch-input-key attachment-data]))
+  (dis/dispatch! [:activity-add-attachment dispatch-input-key attachment-data])
+  (dis/dispatch! [:input [:cmail-data :has-changes] true]))
```

Comment:
@thepug you need to use the passed `dispatch-input-key` or this will work only for Cmail and not on mobile when using the entry-edit component.